student.rb

```diff
@@ -11,4 +11,9 @@ def initialize(age, classroom, name: 'Unknown', parent_permission: true)
   def play_hooky
     '¯\\_(ツ)_/¯'.encode('utf-8')
   end
+
+  def set_classroom(classroom) # rubocop:disable Naming/AccessorMethodName
+    @classroom = classroom
+    classroom.students << self unless classroom.students.include?(self)
+  end
```

Comment:
- [ Required ] I think it would be better if you used the `ruby` syntax for `setter` methods, this means you don't have to create a user-defined method to set the `@classroom` instance variable. You can achieve this by refactoring this code as follows.
```ruby
  def classroom=(classroom)
    @classroom = classroom
    classroom.students << self unless classroom.students.include?(self)
 end
```
Once this is done then, you can refactor the `add_student` method in `classroom.rb` as follows
```ruby
    def add_student(student)
      @students << student
      student.classroom = self
  end
```